### ISSUE TITLE
Support proxy

### ISSUE THREAD

**Issue by HSoulat** [vmware/terraform-provider-tanzu-mission-control#73]
### Is your feature request related to a problem? Please describe.

provider vmware/tanzu-mission-control can't be used on devices behind proxies, even if a proxy configuration is defined in environments variables. Firewall logs clearly shows direct access attempts.

As exemple, azurerm provider is working fine with the same proxy configuration.

### Describe the solution you'd like

Proxy support.

**Comment by shobha2626** [comment]
Hi @HSoulat, TMC terraform provider does not read the proxy configuration from environment variables we need to pass it through the terraform script within the cluster resource

```
spec {
    cluster_group = "default" # Default: default
    proxy         = "proxy-name"
  }
```

Please find more details in the document here - https://registry.terraform.io/providers/vmware/tanzu-mission-control/latest/docs/resources/cluster#attach-cluster-with-proxy

**Comment by HSoulat** [comment]
Hi @shobha2626 ,

Sorry for the confusion, I'm not talking about K8s cluster proxy, but proxy setting to reach TMC "endpoint" API : `my-org.tmc.cloud.vmware.com `

**Comment by shobha2626** [comment]
Could you please share the proxy configuration defined in the environments variables? 
1. `echo $TMC_ENDPOINT`
2. `echo $VMW_CLOUD_ENDPOINT`

Any other var you are setting as well

I hope right value is set to the env var `VMW_CLOUD_API_TOKEN`

**Comment by HSoulat** [comment]
I can at least give you a configuration sample. 
Proxy configuration : 
```
HTTP_PROXY : proxy.consto.com:3129
HTTPS_PROXY : http://proxy.consto.com:3129
```
That settings works on azurerm provider without issue.


For Tanzu provider, I'm using the default configuration listed on exemple usage.
```
provider "tanzu-mission-control" {
  endpoint            = "random.tmc.cloud.vmware.com"          # optionally use TMC_ENDPOINT env var
  vmw_cloud_api_token = var.vmw_cloud_api_token # optionally use VMW_CLOUD_API_TOKEN env var
  vmw_cloud_endpoint = "console.cloud.vmware.com"
}
```

**Comment by shobha2626** [comment]
Thank you @HSoulat for opening this issue to support proxy to reach the TMC endpoint. 

This feature is not supported yet, we will add the support soon and release it as part of the upcoming release.

**Comment by shreyassreenivas** [comment]
This issue feature request has been implemented in https://github.com/vmware/terraform-provider-tanzu-mission-control/pull/88

**Comment by shreyassreenivas** [comment]
https://github.com/vmware/terraform-provider-tanzu-mission-control/releases/tag/v1.1.3

The above feature request is made available in the release v1.1.3
